addon/templates/ui/ui-box.hbs

```diff
@@ -1,34 +1,21 @@
-{{yield}}
-
-{{#yield-slot 'image'}}
-  <div class="card-thumbnail">
-    {{yield}}
-  </div>
-{{/yield-slot}}
-
-{{#yield-slot 'title'}}
-  <div class="card-header">
-    <h5>{{yield}}</h5>
-  </div>
-{{/yield-slot}}
-
-
-{{#yield-slot 'content'}}
-  <div class="card-body {{content-modifier}}">
-    {{yield}}
-  </div>
-{{/yield-slot}}
-
-
-{{#yield-slot 'list'}}
-  <ul class="list-group list-group-flush">
-    {{yield}}
-  </ul>
-{{/yield-slot}}
-
-
-{{#yield-slot 'footer'}}
-  <div class="card-footer">
-    {{yield}}
-  </div>
-{{/yield-slot}}
\ No newline at end of file
+<div class="card-header">
+  <h5>
+    {{yield (hash header=(component 'ui/ui-box/ui-box-header'))}}
+  </h5>
+</div>
+
+<div class="card-body {{content-modifier}}">
+  {{yield (hash body=(component 'ui/ui-box/ui-box-body'))}}
+</div>
+
+<div class="card-footer">
```

Comment:
I didn't realize you could actually do this.. This is sweet.

However I think we want the markup of each section in the component that represents the section.

This should make it easier to pass modifier classes and whatnot, in addition to that if for example in this setup we don't "want" a list, it's still going to render the ul wrapped around the section.

Also, in the "sub components (ui-box-footer) we can actually use the attribute `tagName` and `classNames` to set the correct tag and class names... so we shouldn't have to have the actual markup in the file at all...

Reply:
Does this work? I didn't think was supposed to work.